### PR TITLE
Calculator feature m2

### PR DESCRIPTION
**Pull request by Addis-Belete**
This Milestone contains four components:
- App
- Display
- Button
- ButtonPanel

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import ButtonPannel from './ButtonPanel';
+
+function App() {
+  return (
+    <div>
+      <Display />
+      <ButtonPannel />
+    </div>
+  );
+}
```
Comment (oloomoses) [inline src/components/App.js]: - [ ] Kindly use the ES6 arrow function instead of the function keyword

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import Display from './Display';
+import ButtonPannel from './ButtonPanel';
+
+function App() {
+  return (
+    <div>
+      <Display />
+      <ButtonPannel />
+    </div>
```
Comment (oloomoses) [inline src/components/App.js]: - [ ] Kindly use `React.Fragment` short syntax to wrap up the children components ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable react/require-default-props */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
```
Comment (oloomoses) [inline src/components/Button.js]: - [ ] Do not disable eslint on props instead ensure that all props are properly validated

File: src/components/Button.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function Button(props) {
+  return (
+    <button type="button" className="square">
+      {props.name}
+    </button>
+  );
+}
```
Comment (oloomoses) [inline src/components/Button.js]: - [ ] Kindly use the ES6 arrow function instead of the function keyword

File: src/components/Button.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Button.propTypes = {
+  name: PropTypes.string,
+};
```
Comment (oloomoses) [inline src/components/Button.js]: to avoid the prop type eslint error you can use `name: PropTypes.string.required`

**Review by oloomoses — CHANGES_REQUESTED** [review]
## Hello @Addis0943,
**You have done a commendable job on this milestone**

However, there are some changes to be made before you can move forward

- [x] Kindly use `React.Fragment` short syntax to wrap up the children components ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))
- [x] Kindly use the ES6 arrow function instead of the function keyword for the functional components
- [x] Do not disable eslint on props instead ensure that all props are properly validated

**Check the comments below for more details**

**Please make the above change and ask for another review**

Feel free to leave a comment while mentioning me and I will be happy to reply.
### Happy Coding
_Slack: Oloo Moses_

**Comment by Addis-Belete** [comment]
Hello TSE
Thank you for your feedback. I did on all changes requested. Kindly review it.
Regards

File: src/components/App.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+function App() {
+  return (
+    <>
+      <Display />
+      <ButtonPannel />
+    </>
+  );
+}
+
```
Comment (uche-inyama) [inline src/components/App.js]: Kindly use arrow functions to define your components as requested from the last review.

File: src/components/Button.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable react/destructuring-assignment */
+
```
Comment (uche-inyama) [inline src/components/Button.js]: Kindly enable eslint and fix the error flagged by eslint.

File: src/components/Display.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable react/destructuring-assignment */
+import React from 'react';
```
Comment (uche-inyama) [inline src/components/Display.js]: Kindly enable eslint and fix the error flagged by eslint.

File: src/components/ButtonPanel.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+/* eslint-disable class-methods-use-this */
+import React from 'react';
```
Comment (uche-inyama) [inline src/components/ButtonPanel.js]: Kindly enable eslint and fix the error flagged by eslint.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required :recycle: 

### Hi Addis :slightly_smiling_face: 

There were few changes you did not implement from the last review. I have left inline comments.

Kindly implement the changes then request another code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Comment by Addis-Belete** [comment]
Hello TSE
Thank you for your feedback. I did on all changes requested. Kindly review it.
Regards

**Comment by rukundoeric** [comment]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯
  - Issues raised from the previous review are all addressed. 💯 
  

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/),
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Pull request merged**